### ISSUE TITLE
Defining metrics for Tuning hyperparameters in a custom container

### ISSUE THREAD

**Issue by eveTu** [aws/amazon-sagemaker-examples#1920]
Hey, I'm using a **sklearn** estimator with a custom image URI. 
After running the hyperparameters tuning, I get an empty table in "best training job" with the sentence: `Best training job summary data is available when you have completed training jobs that are emiting metrics.`

I defined metrics the relevant metrics : 
`    metric_definitions=[
           {'Name': 'tr_loss', 'Regex': 'Train_loss=(.*?);'},
           {'Name': 'test_mrr', 'Regex': 'test_mrr=(.*?);'},
           {'Name': 'test_recall', 'Regex': 'test_recall=(.*?);'}
        ]`

But I can't find how to emit them to **stdout** to make them available to the tuner. but I can't find how. 

Any advice/suggestion will help.
Thanks in advance

**Comment by Miles1996** [comment]
Hi @eveTu,

I'm having the same problem. How did you manage to resolve the issue?

Thanks,
Miles

**Comment by eveTu** [comment]
Hey @Miles
Yes, I did. 
I just needed to print out those metrics (or write debug logs) 
and define those metrics in the training job.

For example:
if you print the loss like that: 
`train:loss 5.4` 
'train:loss 2.4'

then you need to specify the **print pattern** and the **expected values**, like this: 
`metric_definitions=[ {'Name': "train:loss", "Regex": ".*train:loss ([0-9\\.]+).*"}]`
you should define those metrics in the training job and choose the relevant metric that you want to optimize in the tuner job as well.

I hope this may help you,
Einav

**Comment by sam-kiki-baker** [comment]
My tuning job prints the metric I'm interested in like this:
- val_recall: 0.7777777910232544

I've set the tuner job up using:
```
objective_metric_name='val_recall',
metric_definitions=[{'Name': 'val_recall', 'Regex': f"'val_recall'=(\d+\.\d+)"}],
```

But I keep getting an error:
`Reason: No objective metrics found after running 1 training jobs. Please ensure that the custom algorithm is emitting the objective metric as defined by the regular expression provided.`

I just don't understand what I'm doing wrong. Why can't it understand the objective metric from the logs?
